Лабораторная работа 1

LarisaBotasheva opened the pull request.

LarisaBotasheva committed:
Лабораторная работа 1

frek17 commented on Лабораторная 1/task_1.py:
```diff
@@ -0,0 +1,45 @@
+    Owl = Toy("коричневый", 0.2)
+    Face = Towel('серый', 40, 20)
+    Body = Towel('синий', 120, 60)
+    import doctest
```
Импорт в начале скрипта

frek17 commented on Лабораторная 1/task_1.py:
```diff
@@ -0,0 +1,45 @@
+        if not isinstance(height, int):
+            raise TypeError("Высота мультиварки должна быть типа int")
+        self.volume = volume
+
```
Методы? Посмотрите пример

frek17 replied:
Описание?

frek17 commented:
Так, пока ничего не буду ставить. Нужны описания, методы, тестирование. Может быть, какие-то вопросы есть?
Все правки вносим в этот PR

LarisaBotasheva committed:
Исправленный task_1

Здравствуйте! Я правильно поняла какие методы нужно было добавлять?
Подскажите, пожалуйста, какой курс я могу рассмотреть по Python, чтобы лучше в нем разбираться. У меня ощущение, что я упустила базовые моменты и поэтому теперь не всегда понимаю что происходит....

frek17 commented on Лабораторная 1/task_1.py:
```diff
@@ -10,6 +12,15 @@ def __init__(self, settings_amount: int, volume: float, height: int):
         if not isinstance(height, int):
             raise TypeError("Высота мультиварки должна быть типа int")
         self.volume = volume
+    def is_empty_MultiCooker(self) -> bool:
```
`_MultiCooker` можно не писать, потому что мы уже внутри этого класса

frek17 commented on Лабораторная 1/task_1.py:
```diff
@@ -31,15 +51,24 @@ def __init__(self, colour_towel: str, length: int, width: int):
+            raise TypeError("Количество раз, которые вы стираете полотенца должно быть целочисленным")
+        if amount_of_times < 0:
+            raise ValueError("Если < 0, то вы не стирали полотенце")
+    def throw_away_the_towel(self, times: int) -> None:
```
`the` лучше не писать

frek17 commented on Лабораторная 1/task_1.py:
```diff
@@ -31,15 +51,24 @@ def __init__(self, colour_towel: str, length: int, width: int):
+        ...
+        
 if __name__ == "__main__":
     Redmond = MultiCooker(7, 4.5, 25)
```
Экземпляры классов пишутся с маленькой буквы

frek17 commented:
Уже лучше. Если что, пингуйте меня, если я в течение нескольких дней не посмотрел. Нотификация может потеряться.

Могу порекомендовать 3 курса по Python и 1 по C++:
https://stepik.org/course/67/syllabus?auth=login - совсем базовый Python
https://stepik.org/course/512/syllabus - уже посложнее Python, достаточно сложных слов, но если прям постараться вникнуть, будет хорошо
https://stepik.org/course/4852/syllabus?auth=login - Pandas (работа с таблицами в Python) и машинное обучение

https://stepik.org/course/363/syllabus?auth=login - понятная подача и практика решения задач на C++
